web/pgadmin/tools/search_objects/static/js/index.js

```diff
@@ -56,7 +56,10 @@ export default class SearchObjectModule {
     let isEnabled = (() => {
       if (!_.isUndefined(obj) && !_.isNull(obj)) {
         if (_.indexOf(pgAdmin.unsupported_nodes, obj._type) == -1) {
-          if (obj._type == 'database' && obj.allowConn) {
+          let t = pgBrowser.tree,
+            selected_item = t.selected(),
```

Comment:
if we're finding the selected item then what is the use of param obj. obj is the selected tree item right?

Reply:
obj is not always selected item. In case of double click on serve node, selected item is server but as we expand server, obj gets values of all db objects listed below server.

Reply:
Then you should probably check the source of obj. obj is intended to check for particular object when selected.